falcon/hooks.py

```diff
@@ -28,8 +39,14 @@
     r'^on_({})(_\w+)?$'.format('|'.join(method.lower() for method in COMBINED_METHODS))
 )
 
+SynchronousResource = Callable[..., Any]
+AsynchronousResource = Callable[..., Awaitable[Any]]
+Resource = Union[SynchronousResource, AsynchronousResource]
+
 
-def before(action, *args, is_async=False, **kwargs):
+def before(
+    action: Resource, *args: Any, is_async: bool = False, **kwargs: Any
```

Comment:
uh ok, I guess a class is accepted by `Callable[..., Any]`.

should probably work as is